### PR TITLE
Remove pyinotify dependency

### LINKED ISSUE #3934: Recommend dropping pyinotify as a dependency
st2 has a dependency on pyinotify, which is poorly maintained. Digging into it, it looks like no code in st2 references it, only the linux pack, which is no longer part of this code base.

Recommend moving to another, dropping it completely or using
https://github.com/chrisjbillington/inotify_simple

pyinotify is also a requirement in @Kami's custom build of logshipper git+https://github.com/Kami/logshipper.git@stackstorm_patched#egg=logshipper

### PR DESCRIPTION
**Pull request by enykeev**
Resolves #3934

### CODE REVIEW

**Comment by Kami** [comment]
What's our plan for this?

Just removing this dependency without any other changes will break linux pack (file watch sensor) out of the box, so we shouldn't do that, imo.

**Comment by Kami** [comment]
If we want to remove it "correctly", we need to update installer scripts / package post install script to correctly create virtual environment for those packages (IIRC, we don't do that right now).

We have also been going back and forth on this for a while now - at first, core and linux packages were not part of standard installation package and they were installed from Exchange, now they are.

In theory, it shouldn't matter how those core / default packages are installed, as long as it's transparent to the end user and packages are always installed no matter what installation method user uses.

/cc @LindsayHill

**Comment by enykeev** [comment]
> Just removing this dependency without any other changes will break linux pack (file watch sensor) out of the box, so we shouldn't do that, imo.

I should have pointed out that the only reason for this PR was to check if the change triggers something in our CI process. 

> we need to update installer scripts / package post install script to correctly create virtual environment for those packages

yep, we need to. I don't really see why core or linux packs should be any different from other packs.

**Comment by enykeev** [comment]
> we need to update installer scripts / package post install script to correctly create virtual environment for those packages

Actually, in light of recent discussion in regards to the future of pack management, I'd say that core and linux pack should be also installed via pack management routine i.e. via API.

**Comment by armab** [comment]
> In short - I am for removing this dependency, but only if and when we found better way to mange installation of it (e.g. virtualenv creation for {core,linux} in post install or similar).
> package post install script to correctly create virtual environment for those packages (IIRC, we don't do that right now)

This will bring a regression in deb/rpm installation itself.
Adding an external dependency pulling something dynamic like that and doing compilation is a bad practice in deb/rpm pakaging. Instead of that, all external resources should be prepared during the package build stage, that's why we went from old deb/rpm to deb/rpm with `dh_virtualenv` with pre-installed pip dependencies on build stage.
Additionally, installing a package is not idempotent and potentially could fail in any step, see #3052 and relies on remote ref/resource.

> yep, we need to. I don't really see why core or linux packs should be any different from other packs.

Disagree.
Having core and linux packs located/shipped externally will add additional installation step and so complicate the story and UX.
Ideally if user has the "core" StackStorm system ready to work right after installing deb/rpm packages.

Add here air-gapped installation requirements, where users don't have immediate access to StackStorm exchange-like resources.
In short, the easy check is: installing `deb`/`rpm` shouldn't rely on any external resource, except of deb/rpm-like dependencies and shouldn't perform any type of compilation/execution, otherwise we'll return back to era with installation issues StackStorm had before.

**Comment by m4dcoder** [comment]
@enykeev @armab @Kami Can you guys resolve your differences here?

**Comment by blag** [comment]
Superceded by #4145.

**Comment by armab** [comment]
:+1:

**Comment by blag** [comment]
And #4145 was superseded by #4191.
